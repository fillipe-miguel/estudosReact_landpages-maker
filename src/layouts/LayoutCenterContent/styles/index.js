import styled, { css } from 'styled-components';

export const LayoutCenterContentRoot = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
  `}
`;

export const InnerHtml = styled.div`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.large};
    text-align: justify;

    & h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      text-align: center;
    }
  `}
`;

export default LayoutCenterContentRoot;
