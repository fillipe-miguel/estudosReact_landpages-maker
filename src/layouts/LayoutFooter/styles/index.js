import styled, { css } from 'styled-components';

import { TextComponentStyles } from '../../../components';
import { SectionContainerStyles } from '../../../sections';

export const LayoutFooterRoot = styled.footer`
  ${({ theme }) => css`
    min-height: 8rem;
    text-align: center;
    background-color: ${theme.colors.white};
    border-top: 1px solid ${theme.colors.mediumGray};

    & ${TextComponentStyles.TextComponentRoot} {
      font-size: ${theme.fonts.sizes.small};

      & p {
        margin: 0;
      }
    }

    & ${SectionContainerStyles.SectionContainerRoot} {
      padding-top: 2rem;
      padding-bottom: 2rem;
    }
  `}
`;

export default LayoutFooterRoot;
