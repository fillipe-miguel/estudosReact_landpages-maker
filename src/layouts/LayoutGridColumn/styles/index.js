import styled, { css } from 'styled-components';

import { HeadingStyles } from '../../../components';

export const LayoutGridColumnRoot = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;

    gap: ${theme.spacings.xxlarge};

    & ${HeadingStyles.HeadingRoot} {
      margin-bottom: ${theme.spacings.small};
    }
  `}

  // ================================= Medias =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      flex-direction: column;

      & ${HeadingStyles.HeadingRoot} {
        margin-bottom: ${theme.spacings.xlarge};
      }
    }
  `}
`;

export const TextContainer = styled.div`
  ${() => css`
    max-width: 35rem;
  `}

  // ================================= Medias =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      min-width: auto;
      text-align: center;
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const ImageContainer = styled.div`
  ${() => css`
    max-width: 64rem;
  `}
`;

export const Image = styled.img`
  ${() => css`
    max-width: 100%;
  `}
`;
