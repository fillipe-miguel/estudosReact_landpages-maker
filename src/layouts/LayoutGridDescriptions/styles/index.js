import styled, { css } from 'styled-components';

import { HeadingStyles } from '../../../components';

export const HeaderContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.xxlarge};

    @media ${theme.medias.lteMedium} {
      text-align: center;
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(270px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.medias.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    & ${HeadingStyles.HeadingRoot} {
      position: relative;
      width: fit-content;
      left: 5rem;
      margin: ${theme.spacings.medium} 0;

      @media ${theme.medias.lteMedium} {
        left: 2rem;
        margin: ${theme.spacings.medium} auto;
      }
    }

    & ${HeadingStyles.HeadingRoot}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.fonts.sizes.xxxlarge};
      left: -5rem;
      top: -2.5rem;
      transform: skew(-15deg);
    }
  `}
`;
