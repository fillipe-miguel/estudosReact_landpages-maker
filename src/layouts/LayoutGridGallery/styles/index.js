import styled, { css } from 'styled-components';

export const HeaderContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.xxlarge};

    @media ${theme.medias.lteMedium} {
      text-align: center;
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(270px, 1fr));
    gap: ${theme.spacings.medium};

    @media ${theme.medias.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${() => css`
    overflow: hidden;
    height: fit-content;

    &:hover > img {
      transition: all 0.3s ease-in-out;
      transform: scale(1.2) rotate(9deg);
    }

    img {
      pointer-events: none;
      max-width: 100%;
    }
  `}
`;
