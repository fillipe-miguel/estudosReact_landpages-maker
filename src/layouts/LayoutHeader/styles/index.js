import styled, { css } from 'styled-components';

import { HeadingStyles } from '../../../components';
import { SectionContainerStyles } from '../../../sections';

export const LayoutHeaderRoot = styled.header`
  position: fixed;
  z-index: 5;
  top: 0;
  right: 0;
  left: 0;

  & > ${SectionContainerStyles.SectionContainerRoot} {
    padding-top: ${({theme}) => theme.spacings.small};
    padding-bottom: ${({theme}) => theme.spacings.small};;
  }

  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    border-bottom: 1px solid ${theme.colors.mediumGray};
  `}

  // ================================= MEDIAS  =================================
  ${({ theme, ...props }) => css`
    @media ${theme.medias.lteMedium} {
      bottom: 0;
      height: 100vh;
      visibility: ${props.visible ? 'visibility' : 'hidden'};
      opacity: ${props.visible ? '1' : '0'};
      transition: all 0.3s ease-in-out;

      & > ${SectionContainerStyles.SectionContainerRoot} {
        display: grid;
        align-items: center;

        height: 100%;
        margin: 0;

        overflow-y: auto;
      }

      & ${HeadingStyles.HeadingRoot} {
        padding-bottom: ${theme.spacings.large};
      }
    }
  `}
`;

export const HeaderContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;

  // ================================= MEDIAS  =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      display: flex;
      flex-direction: column;
      text-align: center;
      padding: ${theme.spacings.xxlarge};
    }
  `}
`;

export const ToggleMenuButton = styled.button`
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;

  display: none;

  font-size: 2.5rem;
  padding: 0.5rem;

  cursor: pointer;
  border: none;

  ${({ theme }) => css`
    background-color: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
  `}

  // ================================= MEDIAS  =================================
  ${({ theme }) => css`
    @media ${theme.medias.lteMedium} {
      display: flex;
      align-items: center;
    }
  `}
`;

export default LayoutHeaderRoot;
