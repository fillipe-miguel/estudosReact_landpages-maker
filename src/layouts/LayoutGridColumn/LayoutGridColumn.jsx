import P from 'prop-types';
import React from 'react';

import * as Styled from './styles';

import { Heading, TextComponent } from '../../components';
import { SectionBackground } from '../../sections';

function LayoutGridColumn({ title, text, setBackground, srcImage }) {
  return (
    <SectionBackground setBackground={setBackground}>
      <Styled.LayoutGridColumnRoot>
        <Styled.TextContainer>
          <Heading
            colorDarker={!setBackground}
            size="xxxlarge"
            level="2"
            uppercase
          >
            {title}
          </Heading>
          <TextComponent colorDarker={!setBackground}>{text}</TextComponent>
        </Styled.TextContainer>

        <Styled.ImageContainer>
          <Styled.Image src={srcImage} alt={title} />
        </Styled.ImageContainer>
      </Styled.LayoutGridColumnRoot>
    </SectionBackground>
  );
}

LayoutGridColumn.defaultProps = {
  setBackground: false,
};

LayoutGridColumn.propTypes = {
  text: P.string.isRequired,
  title: P.string.isRequired,
  srcImage: P.string.isRequired,
  setBackground: P.bool,
};

export default LayoutGridColumn;
