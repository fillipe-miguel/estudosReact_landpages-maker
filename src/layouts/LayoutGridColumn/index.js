import LayoutGridColumn from './LayoutGridColumn';

export default LayoutGridColumn;
