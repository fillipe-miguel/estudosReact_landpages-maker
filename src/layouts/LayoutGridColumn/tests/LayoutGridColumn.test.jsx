/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react';

import LayoutGridColumn from '..';

import * as mocks from '../../../mocks';
import { MyThemeProvider, theme } from '../../../styles';

describe('<LayoutGridColum />', () => {
  // ================================= Teste 01 =================================
  it('Should render without background', () => {
    const { container } = render(
      <MyThemeProvider>
        <LayoutGridColumn {...mocks.layouts.layoutGridColumn} />
      </MyThemeProvider>,
    );

    expect(container.firstChild).toHaveStyle({
      'background-color': theme.colors.white,
    });

    expect(
      screen.queryByRole('heading', { name: 'Grid Two columns' }),
    ).toBeInTheDocument();
  });

  // ================================= Teste 02 =================================
  it('Should render with background', () => {
    const { container } = render(
      <MyThemeProvider>
        <LayoutGridColumn {...mocks.layouts.layoutGridColumn} setBackground />
      </MyThemeProvider>,
    );

    expect(container.firstChild).toHaveStyle({
      'background-color': theme.colors.primaryColor,
    });
  });

  // ================================= Teste 03 =================================
  it('should match snapshot', () => {
    const { container } = render(
      <MyThemeProvider>
        <LayoutGridColumn {...mocks.layouts.layoutGridColumn} setBackground />
      </MyThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
