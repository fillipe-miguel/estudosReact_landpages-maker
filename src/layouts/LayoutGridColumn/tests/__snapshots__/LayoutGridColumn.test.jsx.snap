// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutGridColum /> should match snapshot 1`] = `
.c5 {
  color: #FFF;
  text-transform: uppercase;
  margin-bottom: 3.2rem;
  font-size: 6.4rem;
}

.c6 {
  font-size: 1.6rem;
  color: #FFF;
}

.c6 p {
  margin: 2.4rem 0;
}

.c6 a {
  font-weight: bolder;
  color: #DC143C;
}

.c6 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c6 h1,
.c6 h2,
.c6 h3,
.c6 h4,
.c6 h5,
.c6 h6 {
  margin: 1.6rem 0;
}

.c6 ol,
.c6 ul {
  margin: 1.6rem;
}

.c6 ol > li,
.c6 ul > li {
  margin: 1.6rem 0;
  font-style: italic;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 4.8rem;
}

.c2 .c4 {
  margin-bottom: 1.6rem;
}

.c3 {
  min-width: 35rem;
}

.c7 {
  max-width: 74rem;
}

.c8 {
  max-width: 100%;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  width: 100%;
  padding: 3.2rem;
}

.c0 {
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFF;
  color: #0A1128;
  background-color: #0A1128;
  color: #FFF;
}

@media (max-width:768px) {
  .c5 {
    font-size: 4rem;
  }
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
  }

  .c2 .c4 {
    margin-bottom: 4rem;
  }
}

@media (max-width:768px) {
  .c3 {
    min-width: auto;
    text-align: center;
    margin-bottom: 3.2rem;
  }
}

@media (max-width:768px) {
  .c1 {
    padding-inline: 3.2rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h2
          class="c4 c5"
        >
          Grid Two columns
        </h2>
        <p
          class="c6"
        >
          Teste
        </p>
      </div>
      <div
        class="c7"
      >
        <img
          alt="Grid Two columns"
          class="c8"
          src="public/assets/images/javascript.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
